TypeCobol.LanguageServer/JsonRPC/IRPCServer.cs

```diff
@@ -29,5 +29,10 @@ public interface IRPCServer
         /// Write a trace in the server log file
         /// </summary>
         void WriteServerLog(string trace);
+
+        /// <summary>
+        /// An Unhandled Exception handler
+        /// </summary>
+        UnhandledExceptionEventHandler Handler { get; set; }
```

Comment:
Maybe use an event instead of a delegate property ? It works but an event is more descriptive of what you want to achieve.
```csharp
event UnhandledExceptionEventHandler UnhandledException;
```

Reply:
event is interresting if we sent events here we just use and handler for propagating already sent exceptions.

Reply:
Ok so the question is: which object shuld be responsible of the handling of those unhandled exceptions ?
If it is the IRPCServer instance itself then a method `void OnUnhandledException(Exception e)` in the interface is appropriate. If the handling should be done externally (and is optional) then an event is best.

Reply:
If you mean that another one than LanguageServer can register itself to listen unhandled exception ok event mechanism shall be used, otherwise no use, only one handler shall be set.

Reply:
No I mean a class different from the implementor of IRPCServer, which is the case here. LanguageServer handles the exceptions of JsonRPCServer.

Reply:
First the fact is that class LanguageServer receives in its constructor an IRPCServer interface instance.
Declaring and even in the interface in IRPCServer, will make JsonRPCServer implementation provides add and remove method and then finally use a UnhandledExceptionEventHandler delegate instance.

So in this case event is the equivalent of the delegate UnhandledExceptionEventHandler, with less code using delegate.